x-pack/legacy/plugins/apm/public/components/app/APMIndicesPermission/index.tsx

```diff
@@ -40,13 +39,16 @@ export const APMIndicesPermission: React.FC = ({ children }) => {
     return null;
   }
 
-  const indicesWithoutPermission = Object.keys(indicesPrivileges).filter(
-    index => !indicesPrivileges[index].read
-  );
-
   // Show permission warning when a user has at least one index without Read privilege,
-  // and he has not manually dismissed the warning
-  if (!isEmpty(indicesWithoutPermission) && !isPermissionWarningDismissed) {
+  // and they have not manually dismissed the warning
+  if (
+    indicesPrivileges &&
+    !indicesPrivileges.has_all_requested &&
```

Comment:
By introducing `has_all_requested` we now have two pieces of logic that must be in sync: `has_all_requested` and `indicesWithoutPermission`. 

If for some reason. `has_all_requested` is `false` whereas `indicesWithoutPermission` is empty, we'll render an empty page.